pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PartitionAssignmentGenerator.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.controller.helix.core.realtime;
+
+import com.linkedin.pinot.common.config.TableConfig;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.helix.ZNRecord;
+
+
+/**
+ * Base class for partition assignment generation across tables of a tenant
+ */
+public abstract class PartitionAssignmentGenerator {
+
+  List<String> _tablesForPartitionAssignment;
+  List<String> _instanceNames;
+  int _nReplicas;
+  Map<String, Integer> _nPartitions;
```

Comment:
I would name maps by the items they represent (in this case _tableToNumPartitions, or something like that)